### PR TITLE
MESOS: avoid recovery for mirrored pods on failover

### PR DESCRIPTION
**Pull request by jdef**
on failover the scheduler attempts to rebuild its internal state from the apiserver's pod registry; it should NOT include mirrored pods since those are not managed by the scheduler.

xref https://github.com/mesosphere/kubernetes-mesos/issues/493

### CODE REVIEW

File: contrib/mesos/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -923,6 +924,10 @@ func (ks *KubernetesScheduler) recoverTasks() error {
 		ks.slaves.checkAndAdd(slaveId, t.Offer.Host())
 	}
 	for _, pod := range podList.Items {
+		if _, isMirrorPod := pod.Annotations[kubelet.ConfigMirrorAnnotationKey]; isMirrorPod {
+			// mirrored pods are never reconciled
```
Comment (sttts) [inline contrib/mesos/pkg/scheduler/scheduler.go]: why? // mirrored pods are not managed by the scheduler

Comment (jdef) [inline contrib/mesos/pkg/scheduler/scheduler.go]: before this patch, recoverTasks lists **all** pods across **all**
namespaces and attempts to reconcile them against it's registry. mirrored
pods (static pods) never pass through the scheduler.

Comment (sttts) [inline contrib/mesos/pkg/scheduler/scheduler.go]: I meant the comment. it doesn't add anything to what is immediate in the code line above. The "why" would be helpful.

Comment (jdef) [inline contrib/mesos/pkg/scheduler/scheduler.go]: got it, will clarify

**Comment by jdef** [comment]
PTAL @sttts @s-urbaniak

**Comment by sttts** [comment]
lgtm

**Comment by s-urbaniak** [comment]
lgtm

**Comment by alex-mohr** [comment]
@k8s-bot test this again please (Apparent flake on #14078)

**Comment by alex-mohr** [comment]
@k8s-bot test this again please (#14074 unable to pull from github)

**Comment by alex-mohr** [comment]
@k8s-bot test this again

**Pull request merged**